### PR TITLE
add centrifugo chart

### PR DESCRIPTION
**Pull request by oleh-ozimok**

### CODE REVIEW

**Comment by oleh-ozimok** [comment]
@k8s-bot CLA signed!

File: stable/centrifugo/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+appVersion: 1.7.3
+description: Centrifugo is a real-time messaging server.
+name: centrifugo
+keywords:
+- centrifugo
+- websocket
+- golang
+home: https://github.com/centrifugal/centrifugo
+maintainers:
+  - name: Oleh Ozimok
```
Comment (unguiculus) [inline stable/centrifugo/Chart.yaml]: Please use Github username.

Could you add an `icon`?

File: stable/centrifugo/values.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  repository: centrifugo/centrifugo
+  tag: 1.7.3
+  pullPolicy: IfNotPresent
+service:
+  name: http
+  type: ClusterIP
+  externalPort: 8000
+  internalPort: 8000
+ingress:
+  enabled: true
```
Comment (unguiculus) [inline stable/centrifugo/values.yaml]: I would not enable ingress by default. The configuration assume there is an Nginx ingress controller. I would not require this by default.

File: stable/centrifugo/values.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    presence: true
+    join_leave: true
+    history_size: 10
+    history_lifetime: 30
+    recover: true
+
+args:
+  - --web=true
+
+resources:
```
Comment (unguiculus) [inline stable/centrifugo/values.yaml]: We started recommending not to specify defaults for resources. This should always be a conscious choice of the user.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for your PR.

Please apply best practices for labels. The `app` label should be `{{ template "name" . }}`. Note that this will also require changes to label selectors.

https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

See examples for reference.
https://github.com/kubernetes/helm/tree/master/docs/examples

File: stable/centrifugo/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.service.externalPort }}
+{{- else if contains "ClusterIP"  .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/centrifugo/templates/NOTES.txt]: `"app={{ template "fullname" . }}"`-> `"app={{ template "name" . }},release={{ .Release.Name }}"`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
@k8s-bot ok to test

File: stable/centrifugo/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.service.externalPort }}
+{{- else if contains "ClusterIP"  .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "name" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (unguiculus) [inline stable/centrifugo/templates/NOTES.txt]: `"app={{ template "name" . }}"` -> `"app={{ template "name" . }},release={{ .Release.Name }}"`

File: stable/centrifugo/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the Redis chart and their default values.
+
```
Comment (unguiculus) [inline stable/centrifugo/README.md]: Table incomplete not not correct.

File: stable/centrifugo/values.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    kubernetes.io/ingress.class: nginx
+    ingress.kubernetes.io/upstream-fail-timeout: 60
+    ingress.kubernetes.io/upstream-max-fails: 2
+    ingress.kubernetes.io/proxy-read-timeout: 3600
+    ingress.kubernetes.io/proxy-send-timeout: 3600
+
+config:
+  secret: secret
+  web: true
+  admin_password: password
```
Comment (unguiculus) [inline stable/centrifugo/values.yaml]: Can you provide the option to use a secret for the password?

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
@k8s-bot ok to test

**Pull request merged**